aggregator/aggregator.py

```diff
@@ -134,15 +154,20 @@ def aggregate_data():
     df["Sex"] = df["Sex"].str.capitalize()
     df["AgeGroup"] = df["AgeGroup"].str.strip()
     
+    # Aggregating data
     aggregated = df.groupby([
-        "OccurrenceYear", "Jurisdiction", "Sex", "AgeGroup"
+        "OccurrenceYear", "Jurisdiction", "Sex", "AgeGroup", "DoseCount"
     ], as_index=False).agg(
-        DoseCount=("DoseCount", "sum"),
         Count=("DoseCount", "count")
     )
     
+    # Ensuring ReferenceDate is always 31st December of the OccurrenceYear
+    aggregated["ReferenceDate"] = aggregated["OccurrenceYear"].apply(lambda year: f"{year}-12-31")
```

Comment:
I question if this is correct, at least to my understanding of a reference date.

For the current year at least it is not correct, we should either show the request date for in year or just not provide in-year counts at all.

For past years, it kind of implies the counts for those years are locked down and not subject to change, locked as of the provided reference date, but that won't be guaranteed when live aggregating from the system. Setting the reference date to the date the data was pulled from the system is so that reports produced on the data that pulled on different days have any potential count differences explained by the different data extraction dates.

Reply:
I'll repeat my earlier request, please invite me to these review conversations with Michelle. It will be more fruitful than you and I having proxy-discussions about requirement interpretation.